rocAL/rocAL_pybind/rocal_pybind.cpp

```diff
@@ -316,6 +332,162 @@ namespace rocal{
             .def_readwrite("decode_time",&TimingInfo::decode_time)
             .def_readwrite("process_time",&TimingInfo::process_time)
             .def_readwrite("transfer_time",&TimingInfo::transfer_time);
+        py::class_<rocalTensor>(m, "rocalTensor")
+            .def(
+                "max_shape",
+                [](rocalTensor &output_tensor) {
+                    return output_tensor.info().max_shape();
+                },
+                R"code(
+                Returns a tensor buffer's width.
+                )code"
+            )
+            .def(
+                "batch_size",
+                [](rocalTensor &output_tensor) {
+                    return output_tensor.info().dims().at(0);
+                },
+                R"code(
+                Returns a tensor batch size.
+                )code"
+            )
+            .def("layout", [](rocalTensor &output_tensor) {
+                return rocALToPybindLayout[(int)output_tensor.info().layout()];
+            },
+                R"code(
+                Returns layout of tensor.
+                )code"
+            )
+            .def("dtype", [](rocalTensor &output_tensor) {
+                return rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of tensor.
+                )code"
+            )
+            .def("torch_dtype", [](rocalTensor &output_tensor) {
+                return "torch." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of torch tensor.
+                )code"
+            )
+            .def("numpy_dtype", [](rocalTensor &output_tensor) {
+                return "np." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of numpy array.
+                )code"
+            )
+            .def("cupy_dtype", [](rocalTensor &output_tensor) {
+                return "cp." + rocALToPybindOutputDtype[(int)output_tensor.info().data_type()];
+            },
+                R"code(
+                Returns dtype of cupy array.
+                )code"
+            )
+            .def("dims", [](rocalTensor &output_tensor) {
+                return output_tensor.info().dims();
+            },
+                R"code(
+                Returns dims of tensor.
+                )code"
+            )
+            m.def("dimensions", []=(rocalTensor &output_tensor) {
+            py::list list;
+            for (uint i = 0; i < output_tensor.info().dims(); i++)
```

Comment:
For loop can be avoided here
What is the difference between dims API and dimensions what purpose it was introduced?